libensemble/tests/unit_tests/test_libE_main.py

```diff
@@ -226,7 +226,12 @@ def test_checking_inputs_single():
 
     check_inputs(libE_specs=libE_specs)
     check_inputs(alloc_specs=alloc_specs)
-    check_inputs(sim_specs=sim_specs)
+    try:
+        check_inputs(sim_specs=sim_specs)
+    except AssertionError:
+        assert 1, "Fails because sim_specs['in']=['x_on_cube'] and that's not an 'out' of anything"
+    else:
+        assert 0, "Should have failed"
```

Comment:
Is there a possibility a user may want to perform a self-check of only their sim_specs for consistency, correct fields, etc. ? as in, like in the test, ``check_inputs(sim_specs=sim_specs)``. In which case I think this causes "valid" sim_specs to fail. Maybe I just don't understand the change, but to me it looks like if there's only one dictionary then it checks that the 'in' and 'out' fields match, which isn't a realistic expectation for a single dictionary. Thoughts?

Reply:
I see what you are saying, but issue #382 requested that `check_inputs` checked `sim_specs['in']`against other `['out']` fields. So I think that resolving the issue would break such a usage of `check_inputs` for single sim_specs checks.

Thoughts @shuds13 ?

Reply:
That's fair. Since a user has to at least define both ``sim_specs`` and ``gen_specs`` anyway, it's reasonable to expect they'd want to check both simultaneously for consistency, and also (with this PR) interoperability. Thanks!